### ISSUE TITLE
New vagrant version messing up with authentication

### ISSUE THREAD

**Issue by ssasso** [ipspace/netlab#820]
vagrant version updated via APT repository (new version: 2.3.6-1), plugin vagrant-libvirt forced to 
```
vagrant plugin install vagrant-libvirt --plugin-version=0.11.2
```
is messing up with vagrant insecure key ssh authentication (and keeps asking for device password).

(side note: same behaviour with plugin version 0.12.1, so this is not related to the libvirt plugin)

This means any new user installing a fresh vagrant from APT may suffer of the same problem, I guess.

```
==> sw: Waiting for machine to boot. This may take a few minutes...
DEBUG ssh: Checking key permissions: /root/.vagrant.d/insecure_private_key
 INFO interface: detail: SSH address: 192.168.121.5:22
 INFO interface: detail:     sw: SSH address: 192.168.121.5:22
    sw: SSH address: 192.168.121.5:22
 INFO interface: detail: SSH username: vagrant
 INFO interface: detail:     sw: SSH username: vagrant
    sw: SSH username: vagrant
 INFO interface: detail: SSH auth method: private key
 INFO interface: detail:     sw: SSH auth method: private key
    sw: SSH auth method: private key
DEBUG ssh: Checking key permissions: /root/.vagrant.d/insecure_private_key
 INFO ssh: Attempting SSH connection...
 INFO ssh: Attempting to connect to SSH...
 INFO ssh:   - Host: 192.168.121.5
 INFO ssh:   - Port: 22
 INFO ssh:   - Username: vagrant
 INFO ssh:   - Password? false
 INFO ssh:   - Key Path: ["/root/.vagrant.d/insecure_private_key"]
DEBUG ssh:   - connect_opts: {:auth_methods=>["none", "hostbased", "keyboard-interactive", "publickey"], :config=>false, :forward_agent=>false, :send_env=>false, :keys_only=>true, :verify_host_key=>:never, :password=>nil, :port=>22, :timeout=>15, :user_known_hosts_file=>[], :verbose=>:debug, :logger=>#<Logger:0x00007f373c3796f8 @level=0, @progname=nil, @default_formatter=#<Logger::Formatter:0x00007f373c379658 @datetime_format=nil>, @formatter=nil, @logdev=#<Logger::LogDevice:0x00007f373c379568 @shift_period_suffix=nil, @shift_size=nil, @shift_age=nil, @filename=nil, @dev=#<StringIO:0x00007f373c379770>, @binmode=false, @mon_data=#<Monitor:0x00007f373c379518>, @mon_data_owner_object_id=51760>>, :keys=>["/root/.vagrant.d/insecure_private_key"], :remote_user=>"vagrant", :keepalive=>true, :keepalive_interval=>5}

Password:
```

Using the key directly from SSH works fine.
```
ssh -i /root/.vagrant.d/insecure_private_key vagrant@192.168.121.5
Warning: Permanently added '192.168.121.5' (ECDSA) to the list of known hosts.
localhost#
localhost#
```

Investigation in progress.

**Comment by ssasso** [comment]
(side note: vagrant works well if username/password is used instead of the insecure key)

**Comment by ipspace** [comment]
Shall we pin the version to 2.3.4 in the installation script? Also, do you get the same behavior with other boxes or is this vPTX-specific?

**Comment by ssasso** [comment]
@ipspace yes this is my suggestion as per the above comment.

Any box has the same issue, in fact the one copied above is a vEOS VM.

**Comment by ipspace** [comment]
How far along are you with the vPTX stuff? I could push out 1.5.3-post2 today or wait for your PR and push out 1.5.4.

**Comment by ssasso** [comment]
> How far along are you with the vPTX stuff? I could push out 1.5.3-post2 today or wait for your PR and push out 1.5.4.

Working on the VLAN stuff, which uses different constructs from vMX. I think it should be ready for tomorrow, and I can create an additional PR which includes both the version pin and the mac address format.

**Comment by ssasso** [comment]
According to https://github.com/hashicorp/vagrant/issues/13196#issuecomment-1579309225 , this should be fixed by the end of the week.

If you agree, we can wait and perform some tests with the new version, and avoid to lock to the older version, which one day may disappear from the apt repo

**Comment by ipspace** [comment]
So I did a few tests on a brand-new 22.04 VM:

* Vagrant 2.3.6-1 has the authentication issue but recognizes the "old" MAC address format
* Vagrant 2.3.4 starts the virtual machines but does not recognize the "old" MAC address format (AA-AA-AA...)

It looks like Vagrant development is in a bit of a flux at the moment. I would go with a fixed version (and recommend it in the installation guide similarly to what we did with containerlab) until things settle down. @ssasso I guess we should merge that PR and move on.

Don't you love it when stuff that worked "forever" stops working because someone cleaned up the code or fixed inconsistencies?
